### ISSUE TITLE
macos 2024.6.0 some websites stalls if extension is active

### ISSUE THREAD

**Issue by Fischmuetze** [bitwarden/clients#9617]
### Steps To Reproduce

After update to the last client Bitwarden update on macOS 2024.6.0 some websites stalls in Safari if the Bitwarden extension is active.
e.g.

proxmox Web UI, console part finished with script error
Home Assistant website stalls, no response

Parts of the websites doesn't work, stalls, no response if the extension is active
Seems to be script conflicts. No other extensions active.

### Expected Result

no stalls nor blocking if the extension is active

### Actual Result

e.g.
proxmox Web UI, console part finished with script error
Home Assistant website stalls, no response, not clickable

deactivateíng the Bitwarden extension in Safari removes that issues

### Additional Context

Solution: which isn’t one
deactivate the Bitwarden extension in Safari

### Operating System

macOS

### Operating System Version

14.5 Sonoma

### Web Browser

Safari

### Browser Version

 17.5 (19618.2.12.11.6)

### Build Version

2024.6.0 (25275)

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Fischmuetze** [comment]
> Am 13.06.2024 um 16:13 schrieb Bas Rieter ***@***.***>:
> 
> 
> I experience the exact same issue with Home Assistant from Chrome on Windows. Disabled access to all sites and change to "when you click on the extension". Fixed it for now.
> —

This kind of setting "fix" it for the moment but it isn't a fix in principle. It's just a workaround

**Comment by Virtualchronos** [comment]
Same issue with home assistant. The older 2024.5.2 and older versions doesn't have this bug

**Comment by basrieter** [comment]
> Same issue with home assistant. The older 2024.5.2 and older versions doesn't have this bug

There is no way to rollback right?

**Comment by Virtualchronos** [comment]
I managed to rollback on opera by doing this :
download older release => https://github.com/bitwarden/clients/releases
extract inin it self folder
find your specific browser way to manually add a downloaded extension

**Comment by basrieter** [comment]
I think it is related to https://github.com/bitwarden/clients/issues/713

**Comment by giggio** [comment]
I was having the same problem with home manager, no on every dashboard, but some.
I'm on Edge on Linux.
I downloaded the previous version from: https://github.com/bitwarden/clients/releases/tag/browser-v2024.5.2
I backed up the previous directory (latest version) and extracted the zip onto the same directory name. It worked.
On my configuration it is at `~/.config/microsoft-edge/Default/Extensions/jbkfoedolllekgbhcbcoahefnbanhhlh/2024.6.0_0`.

The maintainers are releasing new versions at this exact moment, but as this issue is still open I might need to do this again tonight after the extension is updated.

**Comment by SyntaxDreamer** [comment]
Same issue with home assistant. Windows with Microsoft Edge. The page freezes, and you can't click anything. The CPU goes to 100% until you close the tab. Disabling the extension or permissions to access all sites works. Absolutely horrible because there is no way to turn the auto-updates off on the extension, roll back or even exclude access to a specific sites, which means the extension has to be disabled completely.

**Comment by ExR90** [comment]
Same issue. I provided support a HAR and Events JSON from Edge Dev Console when the issue is happening. I noticed it first while using Home Assistant, but it happens elsewhere too. Only way to get around the issue is to deny BW extension access to the sites or remove the extension.

Edge Version 126.0.2592.56 (Official build) (arm64)
BW Extension Version: 2024.6.0
MacOS 14.5 (23F79) on an M3 cpu

**Comment by jhmoon2000** [comment]
Same Issue here. Chrome version 125~126 have the problem (including Edge). Naver Whale (chromium 124) does not have the problem with same extension.

**Comment by Adedamola-Aina** [comment]
Hello @Fischmuetze 

I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below. 

Thanks!

**Comment by ovidiupruteanu** [comment]
I was able to reproduce on [https://demo.home-assistant.io/](https://demo.home-assistant.io/) by opening the menu in the top left corner and switching between Overview and Energy a few times.

Windows 11 Pro: 22631.3737
Microsoft Edge: 126.0.2592.56
Bitwarden Extension: 2024.6.0

**Comment by basrieter** [comment]
For me this happens when switching to a view with camera images.

**Comment by twistedsanity** [comment]
Another here:
Windows 11 Pro: 22631.3737 Microsoft Edge: 126.0.2592.56 Bitwarden Extension: 2024.6.0

Home assistant unusable unless uninstall extension.

**Comment by danielbrunt57** [comment]
I'm also having this issue with MS Edge Version 126.0.2592.56 & Bitwarden extension (fresh install) & Home Assistant's overview page which has a Reolink Doorbell Camera image which just spins and the page hangs.  
Chrome Version 126.0.6478.61 with the extension installed loaded fine.  
Chrome just updated to Version 126.0.6478.62 and it's still okay...

**Comment by raidolo** [comment]
Same here, Home Assistant has some issues. Windows 10 and Windows 11 behaviour, I don't think it matters the OS version at this point. Bitwarden Extension in Chrome 2024.6. 
Please fix this!

**Comment by danielbrunt57** [comment]
My extension & Chrome & HA is working fine...

**Comment by danielbrunt57** [comment]
> Bitwarden Extension in Chrome 2024.6

Looks like Bitwarden extension 2024.6 in Chrome (& Edge) is the culprit!

**Comment by raidolo** [comment]
How to downgrade the extension in Windows? I've tried to replace the files in the Extension dir but then Chrome said that the extension was corrupted and it reinstalled the 2024.6... any hint?

**Comment by raidolo** [comment]
I managed to install the previous version 2024.5.2 in the Chrome Extensions page, clicking the button "Load unpacked" (you have to enable the developer mode in top right of the extension page).
It works fine now. It's definetely the 2024.6 version that is affected. I also have Brave with an old extention 2024.4.2 and it's not affected by the issue.

**Comment by danielbrunt57** [comment]
Where do you find the older unpacked extension?

**Comment by raidolo** [comment]
> @raidolo Where do you find the older unpacked extension?

In the releases... https://github.com/bitwarden/clients/releases/download/browser-v2024.5.2/dist-chrome-2024.5.2.zip you have to unzip the file somewhere and open that directory with the "Load Unpacked" button

**Comment by basrieter** [comment]
> d" button

That works, but it won't allow Biometric unlock then.

**Comment by nightcoffee** [comment]
same issue
Windows 11 10.0.22631.3737
Microsoft Edge 125.0.2535.92
Open https://[gerrithub.io](https://gerrithub.io/)/

**Comment by hizzd** [comment]
This issue can be reproduced on the self deploy Gitea login page. After checking the console, I found tens of thousands of error messages that keep increasing:

```
safari-web-extension://xxx/content/bootstrap-autofill.js:369

TypeError: globalThis.requestIdleCallback is not a function. (In 'globalThis.requestIdleCallback(this.processMutations, { timeout: 500 })', 'globalThis.requestIdleCallback' is undefined)
```


```
this.handleMutationObserverMutation = (mutations) => {
    if (this.currentLocationHref !== globalThis.location.href) {
        this.handleWindowLocationMutation();
        return;
    }
    if (!this.mutationsQueue.length) {
!!369 line!! >        globalThis.requestIdleCallback(this.processMutations, { timeout: 500 });
    }
    this.mutationsQueue.push(mutations);
};

```

**Comment by basrieter** [comment]
Same issue on https://photoshop.adobe.com/

**Comment by hizzd** [comment]
Looks like there be a related issue:

https://forum.proxmox.com/threads/safari-only-pve-8-2-4-novnc-console-error.148877/

**Comment by Smokey01966** [comment]
Having this issue to clean install of windows 11 all updates installed
Edge Version 125.0.2535.92 (Official build) (64-bit)
Bitwarden version 2024.6.0
Site having issues on Home assistant dashboard (self hosted)

Steps to reproduce load the webpage, issue is intermittent no changes made between when website runs normal and when it hangs. Issue stops when bitwarden extension is disabled.

Notes:
High CPU usage during hang
Hang seems to happen less often when developer window is open (5-7 refreshes to trigger vs 0-3 with it closed) 
Additional reports for this specific use can be found here https://www.reddit.com/r/homeassistant/comments/1dfl6b6/ha_freezing_in_all_browsers_do_you_have_the/


Screen shot of console when the freeze happens

![Console on freeze](https://github.com/bitwarden/clients/assets/82545255/56fefead-a25b-4d72-a443-1085c6f18a46)


Screenshot of the console during a normal load

![consol normal](https://github.com/bitwarden/clients/assets/82545255/e1f5b7b9-0c93-46e9-aa5b-dc9e4efde401)

**Comment by Smokey01966** [comment]
updated edge to latest Version 126.0.2592.56 (Official build) (64-bit) issue persists

**Comment by klingon888** [comment]
Add me to the list... MS Edge latest BW extension version. Enable extension, website hangs and CPU usage spikes above 100 and stays there. Disable and everything works.

**Comment by MisterHighping** [comment]
Same one in every Chromium based browser. Had the issue in Chrome, Edge and Arc. 
Sites that I could confirm that were affected: Proxmox, Home Assistant, outlook.office.com

**Comment by Arbide** [comment]
Me too, last versión of Chrome (126.0.6478.62) and Bitwarden extension v2024.6.0.

Home assistant frezzes.

**Comment by bscheutjens** [comment]
I have the same problem when I try to go to my Home Assistant. For anyone having this issue, there is a workaround without installing an older version. It has been mentioned before by others, but I liked this on the Home Assistant github because it has a screenshot:

https://github.com/home-assistant/frontend/issues/21085#issuecomment-2167946382

**Comment by derailius** [comment]
i have the same problem trying to load home-assistant

**Comment by chowdarygm** [comment]
same issue on edge browser

**Comment by Ramalama2** [comment]
same issue on safari!
Proxmox novnc console doesnt work anymore!

**Comment by cagonzalezcs** [comment]
Hello everyone,

Thank you for providing context on this bug report. This issue is arising as a result of [a change made to how we collect form fields within ShadowDOM elements](https://github.com/bitwarden/clients/pull/9063). 

We have a feature flag that allows falling back to the former method of collection of those elements that has been enabled. If you are experiencing issues, please go to `Settings > Vault > Sync` and sync your vault to update the server config for the extension. 

If you are on a self-hosted solution, we are working on a hot fix to resolve this issue and will provide an update ASAP.

**Comment by stephen-houchen** [comment]
I just tried this solution.  Settings > Vault > Sync resulted in a message saying that sync failed.  However, it does say that the vault was synced today at 12:49:50 PM (Eastern), just a few minutes before I tried to sync manually.  I'm still having the same issue with Home Assistant.

**Comment by ovidiupruteanu** [comment]
I had the “Syncing Failed” issue as well. It got fixed after I changed the extension permissions from "When you click the extension" to "On all sites".

**Comment by cagonzalezcs** [comment]
@stephen-houchen Are you on a self hosted solution or using Bitwarden cloud? 

If you're on a self-hosted instance, you'll need to roll back to the previous version of the extension [by side-loading v2024.5.2](https://github.com/bitwarden/clients/releases/tag/browser-v2024.5.2)

**Comment by stephen-houchen** [comment]
Thank you for that!  Same here, and the problem with HA has been resolved.  (NOTE:  With "on click" configured, the Bitwarden extension was unable to save a change to an item in the vault, too.)

**Comment by chowdarygm** [comment]
> Hello everyone,
> 
> Thank you for providing context on this bug report. This issue is arising as a result of [a change made to how we collect form fields within ShadowDOM elements](https://github.com/bitwarden/clients/pull/9063).
> 
> We have a feature flag that allows falling back to the former method of collection of those elements that has been enabled. If you are experiencing issues, please go to `Settings > Vault > Sync` and sync your vault to update the server config for the extension.
> 
> If you are on a self-hosted solution, we are working on a hotfix to resolve this issue and will provide an update ASAP. Otherwise, for now it would be best to rollback to the previous version of the Bitwarden extension.

Thanks, this helped and issue resolved

**Comment by cagonzalezcs** [comment]
A new version of the extension `v2024.6.1` has been submitted for review on all web extension stores and should become available to download soon. 

This new version should resolve the issues described in this ticket for self-hosted environments. If you are a Bitwarden Cloud user and are still experiencing issues, you can resolve them by manually syncing your vault within the extension.

I'm closing this ticket for now, but please feel free to continue discussion as necessary. If the issue persists, we'll re-open this ticket and investigate.

**Comment by Grinchii** [comment]
Still the same Problem for me using Safari on MacOS. Installed the update, but TrueNAS Web Gui still freezes after putting in the Password. With Firefox on Mac it works fine.

**Comment by Fischmuetze** [comment]
Persistent problems with proxmox UI noVNC after Update 24.6.1

24.6.1 macOS Sonoma, Safari 17.5 


Bitwarden extension enabled
![bitwarden_enabled](https://github.com/bitwarden/clients/assets/19834496/b8b9c263-ba34-41a6-8ad6-546ef47f4fb6)

Bitwarden extension disabled
![bitwarden_disabled](https://github.com/bitwarden/clients/assets/19834496/9770f74b-2794-4959-af98-8361c16d8690)

**Comment by shashank-brdj** [comment]
Hi All,
Just to update this thread.

I have received the **update** for **Bitwarden extension** **Version 2024.6.1** in MS Edge. I have also verified the same over Brave browser.

![image](https://github.com/bitwarden/clients/assets/77077138/67061a57-a161-4adb-b1bc-92d930690907)

I have changed backed the Site Access option to "**On all sites**" and now the issue seems to be resolved.
Thank you Bitwarden team for the workaround and finally providing quick resolution with an update to extension.

Now my Home Assistant and other services UI loads seamlessly like before.

Thanks.

**Comment by Fischmuetze** [comment]
> I have changed backed the Site Access option to "**On all sites**" and now the issue seems to be resolved. Thank you Bitwarden team for the workaround and finally providing quick resolution with an update to extension.
> 
> Now my Home Assistant and other services UI loads seamlessly like before.
> 

Your UIs may work, but overall as written in my last post here - the issue isn't gone!
In this respect a little bit too early to say thank you.

**Comment by cagonzalezcs** [comment]
Re-opening this thread to continue investigation as we're continuing to see reports of this issue.

One thing I'll mention is that Safari has a [separate known issue](https://github.com/bitwarden/clients/issues/9647) with the usage of the `requestIdleCallback` API in our content script. That will present an error within the console for Safari users, and some aspects of collection of dynamically updated form fields will not work with autofill correctly.

That issue shouldn't cause the lockup of the browser that is described in this thread.

**Comment by klingon888** [comment]
Can confirm 2024.6.1 update fixed the freeze issue with Home Assistant website on Edge. No issues with Proxmox either. TQ for the prompt fix.

**Comment by Ramalama2** [comment]
> Persistent problems with proxmox UI noVNC after Update 24.6.1
> 
> 24.6.1 macOS Sonoma, Safari 17.5
> 
> Bitwarden extension enabled ![bitwarden_enabled](https://private-user-images.githubusercontent.com/19834496/340017147-d29900c0-f1ee-4bf7-98c1-6da506500ba5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3MDQ1ODQsIm5iZiI6MTcxODcwNDI4NCwicGF0aCI6Ii8xOTgzNDQ5Ni8zNDAwMTcxNDctZDI5OTAwYzAtZjFlZS00YmY3LTk4YzEtNmRhNTA2NTAwYmE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE4VDA5NTEyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhNzM1NmY3MWQ1ZmUzMzBkZjhhMDMxNGNkMjg4NTRhMmQ0YjJkYTc0NzBiNzkyNmEzZGQyNTA0MTc3OTJiYmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EJThAV9ZUeCnfHHZ1jmZQpEszeqUZ_idNMGG89JT6bw)
> 
> Bitwarden extension disabled ![bitwarden_disabled](https://private-user-images.githubusercontent.com/19834496/340017154-8f7f9b7e-bd52-4d69-a4fc-d58df7a2bf4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3MDQ1ODQsIm5iZiI6MTcxODcwNDI4NCwicGF0aCI6Ii8xOTgzNDQ5Ni8zNDAwMTcxNTQtOGY3ZjliN2UtYmQ1Mi00ZDY5LWE0ZmMtZDU4ZGY3YTJiZjRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE4VDA5NTEyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1ZTNjYTk5MTBiYjMyMTI0ZmQ1ZGZhNTBjZWI0OThiYTFhNTk5YTA4ZjZlZWRjMGQ3NTY1MGIyOGMyODgwZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fxLNwECW5X12I1gy7ifI5sql8TLk2vWSsRHV_JK_d-Q)

I have the same issue with Safari and 24.6.1
So Safari is definitive still broken!

**Comment by marcolaux** [comment]
`TypeError: globalThis.requestIdleCallback is not a function. (In 'globalThis.requestIdleCallback(this.processMutations, { timeout: 500 })', 'globalThis.requestIdleCallback' is undefined)`

Issue in line 4273 of boostrap-autofill-overlay.js

```
 if (!this.mutationsQueue.length) {
  globalThis.requestIdleCallback(this.processMutations, { timeout: 500 });
}
```

> globalThis.requestIdleCallback is not defined

resulting in an endless loop

![image](https://github.com/bitwarden/clients/assets/6410852/16c50811-65df-4ac2-96d9-07b09fa0273f)

**Comment by Write** [comment]
Safari + Bitwarden 2024.6.1 : Proxmox's console shows a red square "Script error" and you can't interact with console.

**Comment by Reginard** [comment]
> Safari + Bitwarden 2024.6.1 : Proxmox's console shows a red square "Script error" and you can't interact with console.

Same here. Bitwarden app updated to Version 2024.6.2 this morning via the App Store but the Safari extension is still on Version 2024.6.1

**Comment by cagonzalezcs** [comment]
The Safari issue is related to https://github.com/bitwarden/clients/issues/9647 and a fix for the issue has been merged into `main`.

Details for that issue are present in the ticket I linked. For now I'm closing this thread as the issue experienced with lockup of ShadowDOM-heavy websites seems to no longer be present.

**Comment by Reginard** [comment]
2024.6.3 fixes the "no script ..." error in MacOS Safari although the extension still shows as2024.6.1

**Comment by Ramalama2** [comment]
> 2024.6.3 fixes the "no script ..." error in MacOS Safari although the extension still shows as2024.6.1

FINALLY! :-)

**Comment by mokahless** [comment]
Not sure if I should open another issue as I'm reading conflicting info on if this is client or server issue and if server, I'm on vaultwarden not bitwarden.

Latest browser version for Firefox, it seems a server update caused the form fill menu I have set to "off" to work as if it was "on." After changing settings to "when icon is selected," the issue went away. But anyway, prior to that the form autofill menu was causing freezing on Proxmox and - most notability in the way that it might not have been as tested - I'm using the free version so get the popup about license upon login. Bitwarden client prevents me from clicking "ok" requiring me to refresh the page.

Basically what I'm saying is that if this freezing is a client issue, then it is still an issue with 24.9.1 in Firefox, specifically with proxmox free version with the webUI of 7.2-7.

**Comment by mejje** [comment]
Very clear repro on https://chatcraft.org/ which slows down to a crawl when typing a couple of sentences in the inputbox, and eventually throws outofmemory error. Time spent in Bitwarden bootstrap-autofill, with every autofill option disabled.

**Comment by Ramalama2** [comment]
bootstrap-autofill errors everywhere.
I love Bitwarden, and im not sure if oder Password manager have the same issues, so i dont want to blame Bitwarden.
But its a little bit annoying :-(

Lets say, i wish that someone fixes those errors.
Btw the new client interface is so ugly, but it has more features....

However, thanks to everyone who works on Bitwarden and hopefully some great Person is going to fix those errors on all websites.

Cheers
